### PR TITLE
Migrating Converting a Static Site to a Dynamic NodeJS Web App

### PR DESCRIPTION
**Pull request by hectorkambow**
static site to node js article
added hero image
image path
minor edits
one link doesnt work testapp

Moved from [this PR](https://github.com/section-io/engineering-education/pull/106)

### CODE REVIEW

**Comment by hectorkambow** [comment]
@louisefindlay23 can you please check the testapp link - I wasn't able to open it

**Comment by louisefindlay23** [comment]
@hectorkambow, I've fixed it in #124 since I don't have write access to the branch. 

I've reformatted the links as code as they're meant to be an example URL (not lead to anything.) My Markdown editor autoformatted them as links.

File: articles/static-site-dynamic-nodejs-web-app/index.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+Finally, on all pages with a header delete the code between the `<header></header>` tags (the same code you copied to the header.ejs partial file) and replace it with `<% include('../partials/header') %>`. Now, you've created your first EJS partial. Repeat the process for any other repetitive pieces of code such as the head and footer sections.
+
+*Small Tip:* If you find it hard to differentiate between your pages and partials since they have the same .ejs file extension then it can be helpful to put an underscore _ in front of the names of partials so _ header.ejs. This is a naming convention that some developers use that can be helpful.
+
+### Rendering EJS Pages
+Now we get to the exciting part, making the server render the EJS pages and partials so you can see them on the front-end.
+
+First, we need to add the EJS Node module to our server. So, in the `server.js` file, add `const ejs = require('ejs');` near the beginning of the file with the other Node modules.
```
Comment (ninjaginja) [inline articles/static-site-dynamic-nodejs-web-app/index.md:79]: 'near the beginning' is nebulous... You should show exactly where the code goes within the context of server.js

File: articles/static-site-dynamic-nodejs-web-app/index.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+Second, we need to tell our Express server to use EJS so underneath the `app.use(express.static('public'));` line add  `app.set('view engine', 'ejs');`.
```
Comment (ninjaginja) [inline articles/static-site-dynamic-nodejs-web-app/index.md:81]: again, it's better to show the entire server.js code to provide context. Perhaps you could show all of server.js and highlight the lines you're talking through in this section?

**Comment by louisefindlay23** [comment]
@ninjaginja, added my server.js file as an example as requested. Is in PR #157 since I don't have write access to the main repo.

**Comment by ninjaginja** [comment]
covered under #157

**Pull request closed without merging**